plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -842,6 +843,13 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
             return new MaterializedViewFreshness(false);
         }
 
+        return getMaterializedViewFreshness(session, refreshStateMap);
+    }
+
```

Comment:
IMO new method doesn't improve readability. You can extract it, but please motivate that it's for the purpose of follow-up change.

I think you should just squash that change with commit that uses this new method

Reply:
I agree too, that it doesn't improve readability. The purpose of this extracted method was more to make it easier to unit test, because we have no way to stub out `getMaterializedViewToken` manually. To have the unit test pass through `getMaterializedViewToken` would require defining many mock classes especially here: 

https://github.com/trinodb/trino/blob/e5f62436103f5a3c52852aaf48ab9ea3bee4f465/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L884

What do you think?

Reply:
It it's required for tests, you can make it package private and annotate `@VisibleForTests`

Reply:
I just added the mock objects to the test. Probably it's preferable to have clunkier tests rather than expose encapsulation